### PR TITLE
Detect Mount Execution With Hidepid Parameter

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
#2655

## Summary
Detect Mount Execution With Hidepid Parameter
hidepid=0 – The old behavior – anybody may read all world-readable /proc/PID/* files (default) --> No monitoring Needed
hidepid=1 – It means users may not access any /proc// directories, but their own. Sensitive files like cmdline, sched*, status are now protected against other users --> No monitoring Needed
hidepid=2 It means hidepid=1 plus all /proc/PID/ will be invisible to other users. This complicates the task of gathering info about running processes.--> Needs Monitoring

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/linux/defense_evasion_mount_execution.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of mount process being executed with hidepid parameter , which can make processes invisible to
+other users from the system. Adversaries using Linux kernel version 3.2+ (or RHEL/CentOS v6.5+ above) can hide
+the process from other users. When hidepid=2 option is executed to mount the /proc filesystem, only the root user
+can see all processes and the logged-in user only see their own process. This provides a defense evasion mechanism for
+the adversaries to hide their process executions from all other commands such as ps, top, top, pgrep and more.
+With the Linux kernel hardening hidepid option all the user has to do is remount the /proc filesystem with the option,
+which can now be monitored and detected.
```
Comment (imays11) [inline rules/linux/defense_evasion_mount_execution.toml:18]:
```suggestion
Identifies the execution of mount process with hidepid parameter, which can make processes invisible to
other users from the system. Adversaries using Linux kernel version 3.2+ (or RHEL/CentOS v6.5+ above) can hide
the process from other users. When hidepid=2 option is executed to mount the /proc filesystem, only the root user
can see all processes and the logged-in user can only see their own process. This provides a defense evasion mechanism for
the adversaries to hide their process executions from all other commands such as ps, top, pgrep and more.
With the Linux kernel hardening hidepid option all the user has to do is remount the /proc filesystem with the option,
which can now be monitored and detected.
```

Comment (imays11) [reply rules/linux/defense_evasion_mount_execution.toml:18]: small edits

**Review by imays11 — APPROVED** [review]
This is a really cool find, left a few changes for the description but approved 👍🏾

**Pull request merged**